Indicators/Covariance.cs

```diff
@@ -0,0 +1,197 @@
+            _referenceSymbol = referenceSymbol;
+            _targetSymbol = targetSymbol;
+
+            _targetDataPoints = new RollingWindow<decimal>(period);
+            _referenceDataPoints = new RollingWindow<decimal>(period);
+
+            _covariance = 0;
+        }
+
+        /// <summary>
+        /// Creates a new Covariance indicator with the specified target, reference,  
+        /// and period values
+        /// </summary>
+        /// <param name="targetSymbol">The target symbol of this indicator</param>
+        /// <param name="period">The period of this indicator</param>
+        /// <param name="referenceSymbol">The reference symbol of this indicator</param>
+        public Covariance(Symbol targetSymbol, Symbol referenceSymbol, int period)
+            : this($"COV({period})", targetSymbol, referenceSymbol, period)
+        {
+        }
+
+        /// <summary>
+        /// Creates a new Covariance indicator with the specified name, period, target and 
+        /// reference values
+        /// </summary>
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="period">The period of this indicator</param>
+        /// <param name="targetSymbol">The target symbol of this indicator</param>
+        /// <param name="referenceSymbol">The reference symbol of this indicator</param>
+        /// <remarks>Constructor overload for backward compatibility.</remarks>
+        public Covariance(string name, int period, Symbol targetSymbol, Symbol referenceSymbol)
+            : this(name, targetSymbol, referenceSymbol, period)
+        {
+        }
+
+        /// <summary>
+        /// Computes the next value for this indicator from the given state.
+        /// 
+        /// As this indicator is receiving data points from two different symbols,
+        /// it's going to compute the next value when the amount of data points
+        /// of each of them is the same. Otherwise, it will return the last covariance
+        /// value computed
+        /// </summary>
+        /// <param name="input">The input value of this indicator on this time step.
+        /// It can be either from the target or the reference symbol</param>
+        /// <returns>The covariance value of the target used in relation with the reference</returns>
+        protected override decimal ComputeNextValue(IBaseDataBar input)
+        {
+            var inputSymbol = input.Symbol;
+            if (inputSymbol == _targetSymbol)
+            {
+                _targetDataPoints.Add(input.Value);
+            }
+            else if (inputSymbol == _referenceSymbol)
+            {
+                _referenceDataPoints.Add(input.Value);
+            }
+            else
+            {
+                throw new ArgumentException("The given symbol was not target or reference symbol");
+            }
+
+            if (_targetDataPoints.Samples == _referenceDataPoints.Samples && _referenceDataPoints.Count > 1)
```

Comment:
Inside `ComputeCovariance` it's also checking for `IsReady`, should only have a single check for this 👍